test/distributed/_composable/test_checkpoint.py

```diff
@@ -143,6 +170,29 @@ def test_random_cpu(self):
         for p1, p2 in zip(net1.parameters(), net2.parameters()):
             self.assertEqual(p1.grad, p2.grad)
 
+    @parametrize("use_reentrant", [True, False])
+    def test_multi_args(self, use_reentrant: bool):
+        """
+        Tests checkpoint for modules with multiple output args and hence
+        multiple backward function input args.
+        """
+        device = torch.device("cpu")
+        net1 = nn.Sequential(
+            MultiOutputModel(device),
+            MultiInputModel(device),
+            MultiOutputModel(device),
+            MultiInputModel(device),
+        )
+        net2 = deepcopy(net1)
+        checkpoint(net2[0], use_reentrant=use_reentrant)
+        checkpoint(net2[2], use_reentrant=use_reentrant)
+        x1 = torch.randn(20, 100, requires_grad=True)
+        x2 = x1.clone()
+        net1(x1).sum().backward()
+        net2(x2).sum().backward()
+        for p1, p2 in zip(net1.parameters(), net2.parameters()):
+            self.assertEqual(p1.grad, p2.grad)
```

Comment:
maybe test a couple iterations just to be safe?

Reply:
I added more iterations in the draft PR on top. I will land this one for now to avoid re-running CI :/